### ISSUE TITLE
OpenRefine adds "_ - " to column names when creating new project from JSON.

### ISSUE THREAD

**Issue by Opennat** [OpenRefine/OpenRefine#1017]
Refine adds "_ - " in front of all column names (i.e. keys in JSON). It happens on import of any JSON files.

**Comment by tfmorris** [comment]
The underscore character is used as a placeholder for empty (ie null) names and the column name is constructed as a concatenation of the names for the path leading to the current object/field (since the names aren't necessarily unique by themselves).

Are you asking for something different to be done?  We've discussed in the past (and I think there might be an issue for it already), removing the root object from the path.  Are you looking for something along those lines or something different?

While brevity is appreciated, including enough words so that we can tell what you're requesting is helpful.

**Comment by Opennat** [comment]
Pardon about so long answer, there was the reason.

Refine importer does not check for empty keys (or it does not work).
Check it out on simple JSON:
```javascript
[
	{
	"key1": "value1",
	"key2": "value2"
	}
]
```
Column names in Refine for it is `_ - key1` and `_ - key2`, respectively.
It is unnecessary to add any placeholders in this case.

**Comment by Opennat** [comment]
Found this issue #524, but not the pull.

**Comment by wetneb** [comment]
Instead of coming up with our own way to encode the JSON structure in column names, we could use the JSON Pointer syntax for that, which is reasonably standard since it is used by JSON Schema: #2201.

**Comment by tfmorris** [comment]
This is related to #710

**Comment by mshd** [comment]
Any update on this issue, is there a fix?

**Comment by wetneb** [comment]
It would be relatively easy to change this behaviour, although it would likely break any JSON recipes, which rely on specific column names. But it is probably okay to introduce this change anyway, as long as we highlight the potential problem in the release log.

**Comment by wetneb** [comment]
This bug is regularly mentioned, for instance here: https://stackoverflow.com/questions/70825573/underscore-and-dash-in-column-names-after-json-import/70830984#70830984.

**Comment by jquartel** [comment]
I'd love to see this improved (in fact I think that stackoverflow ticket was raised by me). While I do now understand the need for the hyphen when descending into object values, the initial "_ -" seems to be redundant and is especially confusing when the root object is a simple collection of non-object values.

**Comment by mkibp** [comment]
I agree there should be some exception for the simple array of objects as the root object case in terms of the column naming, which is very common for CSV-like data, without actually converting it to CSV and while being able to have nested objects.

There doesn't seem to be a JSON exporter that could be broken by such a change anyway right now (and exporting tables to an array of simple objects would also be a useful feature)
